tests/validate/ps_iops_test/src/ps_iops.cpp

```diff
@@ -0,0 +1,226 @@
+    }
+
+    if (++i == cmds.size())
+      i = 0;
+  }
+
+  arg.end = Clock::now();
+  return (std::chrono::duration_cast<ms_t>(arg.end - arg.start)).count();
+}
+
+void runTestThread(const xrt::device& device, const xrt::kernel& hello_world, arg_t& arg)
+{
+  std::vector<xrt::run> cmds;
+
+  for (int i = 0; i < arg.queueLength; i++) {
+    auto bo0 = xrt::bo(device, DATA_SIZE, hello_world.group_id(0));
+    auto bo1 = xrt::bo(device, DATA_SIZE, hello_world.group_id(1));
+    auto bo0_map = bo0.map<int*>();
+    auto bo1_map = bo1.map<int*>();
+    std::fill(bo0_map, bo0_map + COUNT, 0);
+    std::fill(bo1_map, bo1_map + COUNT, 0);
+    bo0.sync(XCL_BO_SYNC_BO_TO_DEVICE, DATA_SIZE, 0);
+    bo1.sync(XCL_BO_SYNC_BO_TO_DEVICE, DATA_SIZE, 0);
+    auto run = xrt::run(hello_world);
+    run.set_arg(0, bo0);
+    run.set_arg(1, bo1);
+    run.set_arg(2, COUNT);
+    cmds.push_back(std::move(run));
+  }
+  barrier.wait();
+
+  double duration = runTest(cmds, arg.total, arg);
+
+  barrier.wait();
+}
+
+int testMultiThreads(const std::string& dev, const std::string& xclbin_fn, int threadNumber, int queueLength, unsigned int total)
+{
+  std::thread threads[threadNumber];
+  std::vector<arg_t> arg(threadNumber);
+
+  xrt::device device(dev);
+  auto uuid = device.load_xclbin(xclbin_fn);
+  auto hello_world = xrt::kernel(device, uuid.get(), krnl.name);
+
+  barrier.init(threadNumber + 1);
+
+  for (int i = 0; i < threadNumber; i++) {
+    arg[i].thread_id = i;
+    arg[i].queueLength = queueLength;
+    arg[i].total = total;
+    threads[i] = std::thread([&](int i){ runTestThread(device, hello_world, arg[i]); }, i);
+  }
+
+  /* Wait threads to prepare to start */
+  barrier.wait();
+  auto start = Clock::now();
+
+  /* Wait threads done */
+  barrier.wait();
+  auto end = Clock::now();
+
+  for (int i = 0; i < threadNumber; i++)
+    threads[i].join();
+
+  /* calculate performance */
+  int overallCommands = 0;
+  double duration;
+  for (int i = 0; i < threadNumber; i++) {
+    if (verbose) {
+      duration = (std::chrono::duration_cast<ms_t>(arg[i].end - arg[i].start)).count();
+      std::cout << "Thread " << arg[i].thread_id
+                << " Commands: " << std::setw(7) << total
+                << std::setprecision(0) << std::fixed
+                << " IOPS: " << (total * 1000000.0 / duration)
+                << std::endl;
+    }
+    overallCommands += total;
+  }
+
+  duration = (std::chrono::duration_cast<ms_t>(end - start)).count();
+  std::cout << "Overall Commands: " << std::setw(7) << overallCommands
+            << std::setprecision(0) << std::fixed
+            << " IOPS: " << (overallCommands * 1000000.0 / duration)
+            << " (" << krnl.name << ")"
+            << std::endl;
+  return 0;
+}
+
+int _main(int argc, char* argv[])
+{
+  if (argc < 2) {
+    usage(argv[0]);
+    throw std::runtime_error("Number of argument should not less than 2");
+  }
+
+  // Command Line Parser
+  sda::utils::CmdLineParser parser;
```

Comment:
We should be using boost::program_options and not a "custom" command line parser.